core/src/main/java/com/datastax/oss/driver/internal/core/cql/DefaultPreparedStatement.java

```diff
@@ -242,4 +278,93 @@ private ResultMetadata(ByteBuffer resultMetadataId, ColumnDefinitions resultSetD
       this.resultSetDefinitions = resultSetDefinitions;
     }
   }
+
+  private static boolean resolveSkipMetadata(
+      String query,
+      ByteBuffer resultMetadataId,
+      ColumnDefinitions resultSet,
+      DriverExecutionProfile executionProfileForBoundStatements) {
+    String resolveMethod =
+        (executionProfileForBoundStatements != null)
+            ? executionProfileForBoundStatements.getString(
+                DefaultDriverOption.PREPARE_SKIP_CQL4_METADATA_RESOLVE_METHOD)
+            : "smart";
+
+    if (resultSet == null || resultSet.size() == 0) {
+      // there is no reason to send this flag, there will be no rows in the response and,
+      // consequently, no metadata.
+      return false;
+    }
```

Comment:
Technically this is not guaranteed but i think this is correct choice anyway.

Reply:
This is how it worked before on both drivers.